### PR TITLE
Add dynamic CCS hosts

### PR DESCRIPTION
**Pull request by nahiluhmot**
@Daniel0524 @AdamEdgett 

Make each `*Interface` a class that accepts a `host` as an argument to `new`. This required adding a `host` prop to a few views so that they can make requests.

### CODE REVIEW

**Comment by AdamEdgett** [comment]
My only issue here is that you have to pass around this `host` prop that doesn't change and is the same for every interface

**Comment by nahiluhmot** [comment]
@AdamEdgett Yeah, it's annoying but it beats global variables IMO. You could pass around the instances of each interface instead.

**Comment by AdamEdgett** [comment]
Tru
:+1: do it up

File: src/js/components/device.jsx [hunk 1]
```diff
@@ -153,6 +158,14 @@ var Device = React.createClass({
   // is scrolling, so if the user is scrolling,
   isMoving: false,
 
+  getStateInterface: function() {
+    if (!(this._stateInterface instanceof StateInterface)) {
+      this._stateInterface = new StateInterface(this.props.host);
+    }
+
+    return _stateInterface;
```
Comment (Daniel0524) [inline src/js/components/device.jsx]: This returns undefined for me

**Pull request merged**